File: python/longest_strictly_increasing_or_strictly_decreasing_subarray_3105.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        #wait I was gonna brute force but the constraints are chill
        # wait fuck it I'll do that anyway
        inc = 1
        dec = 1
        maxl = 1
        for pos in range(len(nums) -1):
            if nums[pos + 1] > nums[pos]:
                inc += 1
                dec =1 # reset
            elif nums[pos + 1] < nums[pos]:
                inc = 1 # reset
                dec += 1 
            else:
                inc = dec = 1
            maxl = max(inc, dec, maxl)
        return maxl
    

File: rust/src/longest_strictly_increasing_or_strictly_decreasing_subarray_3105.rs
use crate::Solution;

impl Solution {
    pub fn longest_monotonic_subarray(nums: Vec<i32>) -> i32 {
        let mut inc = 1;
        let mut dec = 1;
        let mut max = 1;

        for i in 0.. nums.len() - 1 {
            if nums[i+ 1] > nums[i] {
                inc +=1;
                dec = 1;

            }
            else if nums[i + 1] < nums[i] {
                inc = 1;
                dec +=1;

            }

            else { // they are equal :(
                inc = 1;
                dec = 1;
            }
            max = max.max(inc);
            max = max.max(dec);
        }
    max
    }
}
